Sources/Xcodeproj/pbxproj().swift

```diff
@@ -539,17 +540,35 @@ func xcodeProject(
             if fileSystem.isFile(clangTarget.moduleMapPath) {
                 moduleMapPath = clangTarget.moduleMapPath
                 isGenerated = false
+
+                includeGroup.addFileReference(path: moduleMapPath.asString, name: moduleMapPath.basename)
+                // Save this modulemap path mapped to module so we can later wire it up for its dependees.
+                modulesToModuleMap[target] = (moduleMapPath, isGenerated)
             } else {
-                // Generate and drop the modulemap inside Xcodeproj folder.
-                let path = xcodeprojPath.appending(components: "GeneratedModuleMap", clangTarget.c99name)
-                var moduleMapGenerator = ModuleMapGenerator(for: clangTarget, fileSystem: fileSystem)
-                try moduleMapGenerator.generateModuleMap(inDir: path)
-                moduleMapPath = path.appending(component: moduleMapFilename)
-                isGenerated = true
+                let umbrellaHeaderName = clangTarget.c99name + ".h"
+                if includeGroup.subitems.contains(where: { $0.path == umbrellaHeaderName }) {
+                    // If an umbrella header exists, enable Xcode's builtin module's feature rather than generating
+                    // a custom module map. This increases the compatibility of generated Xcode projects.
+                    targetSettings.common.CLANG_ENABLE_MODULES = "YES"
```

Comment:
Is this different from DEFINES_MODULE?

Reply:
Xcode's Quick Help said:
> CLANG_ENABLE_MODULES
--
Boolean
Enables the use of modules for system APIs. System headers are imported as semantic modules instead of raw headers. This can result in faster builds and project indexing.

I am setting this by mimicking the framework target settings generated by Xcode.
I do not know if it is essential to set this to YES, but I have never had any troubles with doing this.

Reply:
> I do not know if it is essential to set this to YES,

I confirmed that this setting is required by testing on https://github.com/jpsim/Yams

Reply:
Ah cool. Can you also implement the `MODULEMAP_FILE` setting that @ddunbar suggested?

Reply:
I'll try.